### PR TITLE
iio: iio.c: Handle return error if no triggers are present

### PR DESCRIPTION
**Pull request by rbolboac**
With the current implementation -EINVAL error is returned when searching for triggers. However, libiio exptects -ENOENT error if no triggers are present.
This commit returns the correct error if no triggers are found.

Signed-off-by: Ramona Bolboaca <ramona.bolboaca@analog.com>

### CODE REVIEW

File: iio/iio.c [hunk 1]
```diff
@@ -926,8 +926,17 @@ static int iio_get_trigger(struct iiod_ctx *ctx, const char *device,
 			   char *trigger, uint32_t len)
 {
 	struct iio_dev_priv *dev;
+	struct iio_trig_priv *trig;
 	struct iio_desc *desc = ctx->instance;
 
+	if (!desc->nb_trigs)
+		return -ENOENT;
+
+	trig = get_iio_trig_device(desc,device);
```
Comment (buha) [inline iio/iio.c:935]: `, `

File: iio/iio.c [hunk 2]
```diff
@@ -954,9 +963,19 @@ static int iio_set_trigger(struct iiod_ctx *ctx, const char *device,
 			   const char *trigger, uint32_t len)
 {
 	struct iio_dev_priv	*dev;
+	struct iio_trig_priv	*trig;
 	uint32_t i;
+	struct iio_desc *desc = ctx->instance;
```
Comment (buha) [inline iio/iio.c:968]: maybe `ctx` could be NULL (not sure, double-check)

Comment (rbolboac) [reply iio/iio.c:968]: the ctx cannot be null with the current implementation...it comes from iiod: struct iiod_ctx ctx = IIOD_CTX(desc, conn);
also, it is never checked in iio.c

Comment (buha) [reply iio/iio.c:968]: ok, fair enough, that's why i asked

**Review by buha — APPROVED** [review]

**Review by amiclaus — APPROVED** [review]

**Review by CiprianRegus — APPROVED** [review]

**Pull request merged**